### PR TITLE
simple-cipher: update tests and add version file #1475

### PR DESCRIPTION
**Pull request by hgvanpariya**
---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by hgvanpariya** [comment]
@FridaTveit : Please provide your review comments.

File: exercises/simple-cipher/.meta/src/reference/java/Cipher.java [hunk 1]
```diff
@@ -17,7 +17,7 @@ public Cipher() {
 
     public Cipher(String key) {
         if (!isValidKey(key)) {
-            throw new IllegalArgumentException("Invalid key");
+            throw new IllegalArgumentException("Bad key");
```
Comment (FridaTveit) [inline exercises/simple-cipher/.meta/src/reference/java/Cipher.java]: I think I prefer "Invalid key" as exception message 🙂

File: exercises/simple-cipher/src/test/java/IncorrectKeyCipherTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Before
+    public void setup() {
+    }
+
+    /**
+     * Throws an error with an all uppercase key
+     */
```
Comment (FridaTveit) [inline exercises/simple-cipher/src/test/java/IncorrectKeyCipherTest.java]: I don't think you need these comments. You should be able to use the test name to convey what it's testing 🙂

File: exercises/simple-cipher/src/test/java/IncorrectKeyCipherTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Step 3: Incorrect key cipher
+ */
```
Comment (FridaTveit) [inline exercises/simple-cipher/src/test/java/IncorrectKeyCipherTest.java]: Having the step number in the test file names might make it easier for users to understand what each file is for? And also the hints.md file refers to those file names 🙂

File: exercises/simple-cipher/src/test/java/SubstitutionCipherTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    }
+    
+    /**
+     * Can double shift encode
+     */
+    @Ignore("Remove to run test")
+    @Test
+    public void cipherCanDoubleShiftEncode() {
+        String plainText = "iamapandabear";
+        String cipherText = "qayaeaagaciai";
```
Comment (FridaTveit) [inline exercises/simple-cipher/src/test/java/SubstitutionCipherTest.java]: I'd prefer to be consistent with having these variables; either have `plainText` and `cipherText` variables for each test or for none of them, if that makes sense? 🙂

**Comment by FridaTveit** [comment]
@hgvanpariya How are you getting on with this? 🙂

**Comment by hgvanpariya** [comment]
@FridaTveit : I'll update it by tomorrow ...

File: exercises/simple-cipher/src/test/java/Step1RandomKeyCipherTest.java [hunk 5]
```diff
@@ -1,13 +1,17 @@
+package java;
```
Comment (Smarticles101) [inline exercises/simple-cipher/src/test/java/Step1RandomKeyCipherTest.java]: I don't think `package java;` is needed?

File: exercises/simple-cipher/src/test/java/Step3IncorrectKeyCipherTest.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
+package java;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertEquals;
```
Comment (Smarticles101) [inline exercises/simple-cipher/src/test/java/Step3IncorrectKeyCipherTest.java]: This import is unused and is causing the build to fail. `Unused import - org.junit.Assert.assertEquals.`

File: exercises/simple-cipher/src/test/java/Step1RandomKeyCipherTest.java [hunk 7]
```diff
@@ -1,13 +1,17 @@
...
 import org.junit.Test;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.regex.Pattern;
 
 /**
- * Step 1: Make a simple shift cipher
+ * Step 1: Random key cipher
```
Comment (Smarticles101) [inline exercises/simple-cipher/src/test/java/Step1RandomKeyCipherTest.java]: Like @FridaTveit said, can you remove all the comments? The names of the tests should explain them well enough.

File: exercises/simple-cipher/src/test/java/SimpleCipherStepThreeTest.java [hunk 8]
```diff
@@ -1,63 +0,0 @@
-import org.junit.Ignore;
```
Comment (Smarticles101) [inline exercises/simple-cipher/src/test/java/SimpleCipherStepThreeTest.java]: Is it necessary to rename the test files? I think for consistency's sake we should stick to the previous names. Thoughts @FridaTveit ?

**Comment by Smarticles101** [comment]
Closing due to inactivity. If you're still working on this @hgvanpariya, feel free to update the PR and I'll open it again! 😄

**Pull request closed without merging**